pkg/controller/registry/registry_controller_reconcilier_test.go

```diff
@@ -0,0 +1,234 @@
+	"github.com/kubic-project/registries-operator/pkg/test"
+	"github.com/kubic-project/registries-operator/pkg/test/fake"
+	"k8s.io/apimachinery/pkg/types"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"golang.org/x/net/context"
+	. "github.com/onsi/gomega"
+	"testing"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+
+type FakeCertReconciler struct {
+	reconcileMissing bool
+	reconcilePresent bool
+}
+
+func NewFakeCertReconciler() *FakeCertReconciler {
+	return &FakeCertReconciler{false, false}
+}
+
+
+func newTestReconcileRegistry()  ReconcileRegistry {
+	return ReconcileRegistry{
+		fake.NewTestClient(),
+		fake.NewTestRecorder(),
+		scheme.Scheme,
+		NewFakeCertReconciler(),
+	}
+}
+
+func (r *FakeCertReconciler) ReconcileCertPresent(registry *kubicv1beta1.Registry,
+	curNodes map[string]*corev1.Node,
+	specSecret *corev1.Secret) (reconcile.Result, error) {
+
+	r.reconcilePresent = true;
+	return  reconcile.Result{}, nil
+}
+
+
+func (r *FakeCertReconciler) ReconcileCertMissing(instance *kubicv1beta1.Registry, nodes map[string]*corev1.Node) error {
+	r.reconcileMissing = true;
+	return  nil
+
+}
+
+func (r FakeCertReconciler) ReconcileCertPresentCalled() bool {
+	return r.reconcilePresent && !r.reconcileMissing
+}
+
+func (r FakeCertReconciler) ReconcileCertMissingCalled() bool {
+	return r.reconcileMissing && !r.reconcilePresent
+}
+
+func (r FakeCertReconciler) ReconcileCertNotCalled() bool {
+	return !r.reconcileMissing && !r.reconcilePresent
+}
+
+//checks if the registry is finalizing
+func (r *ReconcileRegistry)isFinalizing(registry string) (bool, error) {
+
+	instance:= &kubicv1beta1.Registry{}
+	err:= r.Get(context.TODO(),types.NamespacedName{Name: registry}, instance)
+
+	if err == nil {
+		return instance.ObjectMeta.Finalizers[0] == regsFinalizerName, nil
+	} else {
+		return false, err
+	}
+}
+
+func TesRegistryNotFound(t *testing.T) {
+
+	g := NewGomegaWithT(t)
+
+	r := newTestReconcileRegistry()
+
+	req := reconcile.Request{types.NamespacedName{Name: "TestRegistry",Namespace: ""}}
+	res, _ := r.Reconcile(req)
+
+        g.Expect(res).To(Equal(reconcile.Result{}))
+	//neither of reconcile methods should be called
+	cr, _ :=r.certReconciler.(*FakeCertReconciler)
+	g.Expect(cr.ReconcileCertNotCalled()).Should(Equal(true))
+
+}
+
+func TestRegistryFinalizing(t *testing.T) {
+
+	g := NewGomegaWithT(t)
+
+	r := newTestReconcileRegistry()
+
+	fooSec, err := test.BuildSecretFromCert("foo-ca-crt", "foo.crt")
+
+	if err != nil {
+		t.Errorf("Error creating secret %v", err)
+	}
+
+	fooReg, err :=kubicv1beta1.GetTestRegistry("foo")
+	if err != nil {
+		t.Errorf("Error Getting Registry %v", err)
+	}
+	//simulate registry is being deleted
+	timestamp := metav1.Now()
+	fooReg.ObjectMeta.SetDeletionTimestamp(&timestamp)
+
+	//simulate the registry had a certificated deployed
+	fooReg.Status.Certificate.CurrentHash = getSecretHash(fooSec)
+
+	c := r.Client
+	c.Create(context.TODO(), fooSec)
```

Comment:
i'm not sure what this is doing as `fooSec` is not used after this.

Reply:
the Secret is used internally by the reconciliation logic. It may not be used in all test cases, but I tried to unify the setup thinking in refactor it.

Reply:
I think that the setting shouldn't dependent too much of the internal working. That is, maybe I know the object is not (or should not) be used in an specific use case, but that's not part of the specification, it is an implementation detail.

Reply:
:+1: